lib/internal/http2/core.js

```diff
@@ -1479,7 +1480,15 @@ class ClientHttp2Session extends Http2Session {
 
     const onConnect = requestOnConnect.bind(stream, headersList, options);
     if (this.connecting) {
-      this.once('connect', onConnect);
+      if (this[kPendingRequestCalls]) {
+        this[kPendingRequestCalls].push(onConnect);
+      } else {
+        this[kPendingRequestCalls] = [onConnect];
+        this.once('connect', () => {
+          this[kPendingRequestCalls].forEach((f) => f());
+          delete this[kPendingRequestCalls];
```

Comment:
Can you reset to `null` instead? `delete` changes the way an object works in V8 (or at least it did for the longest time), so this would have non-trivial performance impact also.